plugins/defaultplugins/ifplugin/afpacket_config_test.go

```diff
@@ -0,0 +1,555 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ifplugin_test
+
+import (
+	"testing"
+
+	"git.fd.io/govpp.git/adapter/mock"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/cn-infra/utils/safeclose"
+	"github.com/ligato/vpp-agent/idxvpp/nametoidx"
+	ap_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/af_packet"
+	if_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+	"github.com/ligato/vpp-agent/tests/vppcallmock"
+	. "github.com/onsi/gomega"
+)
+
+/* AF_PACKET configurator init */
+
+// Test init function
+func TestAfPacketConfiguratorInit(t *testing.T) {
+	RegisterTestingT(t)
+	connection, err := core.Connect(&mock.VppAdapter{})
+	Expect(err).To(BeNil())
+	plugin := &ifplugin.AFPacketConfigurator{}
+	vppCh, err := connection.NewAPIChannel()
+	Expect(err).To(BeNil())
+	err = plugin.Init(logrus.DefaultLogger(), vppCh, struct{}{}, nil, nil)
+	Expect(err).To(BeNil())
+	connection.Disconnect()
+}
+
+/* AF_PACKET test cases */
+
+// Configure af packet interface with unavailable host
+func TestAfPacketConfigureHostNotAvail(t *testing.T) {
+	var err error
+	// Setup
+	ctx, plugin, _ := afPacketTestSetup(t)
+	defer afPacketTestTeardown(ctx)
+	// Data
+	var addresses []string
+	data := getTestAfPacketData("if1", append(addresses, "10.0.0.1/24"), "host1")
```

Comment:
Replace `append(addresses, "10.0.0.1/24")` with `[]string{"10.0.0.1/24"}` and remove the addresses variable. This applies to all other test cases as well.